src/plugins/intel_npu/src/plugin/npuw/lazy_tensor.cpp

```diff
@@ -68,6 +71,24 @@ struct Const {
         m_read_from_bin = ov::Tensor(m_cached_type, m_cached_shape);
         std::memcpy(m_read_from_bin.data(), weights->get_ptr(m_offset), m_byte_size);
     }
+    void read_weight(const std::shared_ptr<const ov::Model>& model) {
+        NPUW_ASSERT(!m_node &&
+                    "LazyTensor can only read weight when it's being deserialized and not created from a Constant!");
+        m_read_from_bin = ov::Tensor(m_cached_type, m_cached_shape);
+        bool was_read = false;
+        for (const auto& node : model->get_ordered_ops()) {
```

Comment:
it can be very slow to search each constant across the whole graph each time. 
maybe you can create an index first, i.e. a map to map constant ID -> constant and then use in to restore weights.

Reply:
Done